src/data/set/finite.lean

```diff
@@ -994,6 +994,25 @@ by { rw ←range_subset_iff at hf, exact (infinite_range_of_injective hi).mono h
 lemma infinite.exists_nat_lt {s : set ℕ} (hs : s.infinite) (n : ℕ) : ∃ m ∈ s, n < m :=
 let ⟨m, hm⟩ := (hs.diff $ set.finite_le_nat n).nonempty in ⟨m, by simpa using hm⟩
 
+lemma infinite_of_forall_exists_nat_lt {S : set ℕ} (h : ∀ (n : ℕ), ∃ m ∈ S, n < m) : S.infinite :=
+begin
+    intro hS,
+    let S2 : finset ℕ := set.finite.to_finset hS,
+    have h2 : ∃ B, ∀n ∈ S2, n ≤ B,
+    { use finset.sup S2 id,
+      intros,
+      apply finset.le_sup H },
+    cases h2 with N hN,
+    cases h N with n hn,
+    cases hn with hnS hnN,
```

Comment:
Instead of building an existential then taking it apart, you could skip straight to using
```lean
obtain ⟨n, hnS, hnN⟩ := h (finset.sup S2 id)
```
where `obtain` also combines the last two `cases` into one.